src/components/menu-component.jsx

```diff
@@ -68,16 +70,34 @@ function menuComponents(items, parentComponent) {
 class MainMenu extends React.Component {
   constructor(props) {
     super(props)
+    this.state = {
+      isModalOpen: false,
+    }
     this.insertCell = this.insertCell.bind(this)
     this.cellUp = this.cellUp.bind(this)
     this.cellDown = this.cellDown.bind(this)
     this.runCell = this.runCell.bind(this)
     this.deleteNotebook = this.deleteNotebook.bind(this)
     this.runAllCells = this.runAllCells.bind(this)
+    this.switchModal = this.switchModal.bind(this)
```

Comment:
We need a more descriptive name for this, like `this.switchDeleteNotebookDialog`, so it is not ambiguous what the dialog does.